metriq_gym/run.py

```diff
@@ -89,11 +95,60 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     ]
     if all(task.status() == JobStatus.COMPLETED for task in quantum_job):
         result_data: list[ResultData] = [task.result().data for task in quantum_job]
-        print(handler.poll_handler(job_data, result_data))
+        results: BenchmarkResult = handler.poll_handler(job_data, result_data)
+        print(results)
+        if is_upload:
+            upload_job(args, job_type, job_data, results, platforms[metriq_job.device_name.lower()])
     else:
         logger.info("Job is not yet completed. Please try again later.")
 
 
+def upload_job(args: argparse.Namespace, job_type: JobType, job_data: BenchmarkData, result_data: BenchmarkResult, platform: int):
```

Comment:
As we include more benchmarks, it's likely that the `upload_job` function will be come larger by proxy. Do we want to move some of this to another file to exclusively handle upload logic and then import in `run.py` from there?

Reply:
If we do that, I think file should be called something like "actions.py" and include `dispatch_job`, `poll_job`, and `upload_job`, but I think that's out-of-scope for this PR. But, I would move all three or else none of them.

Reply:
This was done.